toolchain/check/import_ref.cpp

```diff
@@ -125,44 +125,58 @@ auto VerifySameCanonicalImportIRInst(Context& context, SemIR::InstId prev_id,
-//      import the constant.
-//    - For instructions that can be forward declared, if we don't already have
-//      a constant value from a previous attempt at resolution, start by making
-//      a forward declared constant value to address circular references.
-//    - Gather all input constants.
+//    - If we've processed this work item before, then we now process it again,
+//      because it didn't complete last time, even though we have a constant
+//      value already.
+//
+// 2. Resolve the instruction (TryResolveInst/TryResolveTypedInst), in up to
+//    three passes. First pass:
+//    - Gather all input constants necessary to form the constant value of the
+//      instruction.
 //      - Gathering constants directly adds unresolved values to work_stack_.
-//    - If any need to be resolved (HasNewWork), return Retry(): this
-//      instruction needs another call to complete.
-//      - If the constant value is already known because we have made a forward
-//        declaration, pass it to Retry(). It will be passed to future attempts
-//        to resolve this instruction so the earlier work can be found, and will
-//        be made available for other instructions to use.
-//      - The subsequent attempt to resolve this instruction must produce the
-//        same constant, because the value may have already been used by
-//        resolved instructions.
-//    - Build any necessary IR structures, and return the output constant.
-// 3. If resolve didn't return Retry(), pop the work. Otherwise, it needs to
-//    remain, and may no longer be at the top of the stack; set `retry` on it so
-//    we'll make sure to run it again later.
+//    - If HasNewWork() reports that any work was added, then return Retry():
+//      this instruction needs another call to complete. Gather the now-resolved
+//      constants and continue to the next step once the retry happens.
 //
-// TryResolveInst can complete in one call for a given instruction, but should
-// always complete within three calls:
+//    Second pass:
+//    - Build the constant value of the instruction.
+//    - Gather all input constants necessary to finish importing the
+//      instruction. This is only necessary for instructions like classes that
+//      can be forward-declared. For these instructions, we first import the
+//      constant value and then later import the rest of the declaration in
+//      order to break cycles.
+//    - If HasNewWork() reports that any work was added, then return
+//      Retry(constant_value): this instruction needs another call to complete.
+//      Gather the now-resolved constants and continue to the next step once the
+//      retry happens.
 //
-// - TryResolveInst can retry once if the generic is not yet loaded.
-// - TryResolveTypedInst can retry once if its inputs are not yet loaded.
-// - The third call should succeed.
+//    Third pass:
+//    - After this point, the constant value for the instruction is already set,
```

Comment:
What does "after this point" mean? After the third pass? After the second pass?

```suggestion
//    - After the second pass, the constant value for the instruction is already set,
```